app/models/diagnosed_need.rb

```diff
@@ -3,6 +3,7 @@
 class DiagnosedNeed < ApplicationRecord
   belongs_to :diagnosis
   belongs_to :question
+  has_many :assistances, through: :question
```

Comment:
I think it is not necessary here. It could bring confusion between "potential assistances" relevant to a specific need, and between "selected assistances" that were chosen during a diagnosis.